### DISCUSSION TITLE
Fetching Data Through Likes Join Table To Know If Liked

### DISCUSSION THREAD

**Discussion by dalezak** [supabase/supabase#13442] [Questions] [answered]
I have a `quotes` and `likes` table, wondering how to fetch `quotes` including whether the user has `liked` each `quote`. 

If I select `*,likes(count)` then I get total number of `likes` for a `quote`, not whether the user `liked` it or not.

If I select `*,likes!inner(*)` then it only returns the `quotes` that the user has `liked`.

Think I'm looking for something like this https://stackoverflow.com/a/71414832/385730.

```sql
select
    q.*
    ,EXISTS(SELECT * FROM likes l WHERE l.quote_id = q.id and l.user_id = 4) AS is_liked
FROM quotes q
order by q.id;
```

Is this possible to fetch `quotes` and `likes` in single query in Supabase? Or would I need to fetch `quotes` and `likes` separately? Or create a view with the two tables combined?

**Comment by steve-chavez** [comment, marked as answer]
I suggest to create a view or function and just use plain SQL there. See https://supabase.com/docs/reference/javascript/rpc.

**Comment by dalezak** [comment]
@steve-chavez I added a function, however it appears to only return a single record? Is this a config issue or limitation in Supabase? I noticed [this issue](https://github.com/orgs/supabase/discussions/4245) reporting the same problem.

**Reply by steve-chavez** [reply]
Yeah I think that will be fixed soon. For now you need to create the function with plain SQL, make sure it has `RETURNS SETOF..`
